packages/teams-js/src/internal/communication.ts

```diff
@@ -180,12 +183,20 @@ export function sendMessageToParent(actionName: string, argsOrCallback?: any[] |
   }
 }
 
+const sendMessageToParentHelperLogger = communicationLogger.extend('sendMessageToParentHelper');
+
 /**@internal */
 function sendMessageToParentHelper(actionName: string, args: any[]): MessageRequest {
+  const logger = sendMessageToParentHelperLogger;
```

Comment:
Can we implement logging levels? This looks to be a lot of logging, maybe these could be divided into a couple of levels to avoid excessive logging? Also, how will this be turned on? Is it by the user, the app developer, or a feature flag?

Reply:
The `debug` package, which is, from what I understand, the gold standard for this kind of logging, does not provide log levels. Instead, it leverages namespaces/categories to allow users to enable/disable certain logs. Nothing will be turned on by default, and the user can either enable all the logging (by setting `localStorage.debug = 'teamsJs.*'`) or e.g. only the communication functions (`localStorage.debug = 'teamsJs:communication:*`) or even just one of them (`localStorage.debug = 'teamsJs:communication:flushMessageQueue`). 

Because this is not getting uploaded anywhere, is off by default, I don't want to engineer a new layer on top of that, unless strictly necessary.

Reply:
That makes sense. It would be good to instrument some ways for the user to control these settings through the Teams app (even if it's just through the url query), rather than having to muck with the console.